crates/polars-time/src/date_range.rs

```diff
@@ -111,25 +111,43 @@ pub(crate) fn datetime_range_i64(
         ComputeError: "`interval` must be positive"
     );
 
-    let size: usize;
-    let offset_fn: fn(&Duration, i64, Option<&Tz>) -> PolarsResult<i64>;
-
-    match tu {
-        TimeUnit::Nanoseconds => {
-            size = ((end - start) / interval.duration_ns() + 1) as usize;
-            offset_fn = Duration::add_ns;
-        },
-        TimeUnit::Microseconds => {
-            size = ((end - start) / interval.duration_us() + 1) as usize;
-            offset_fn = Duration::add_us;
-        },
-        TimeUnit::Milliseconds => {
-            size = ((end - start) / interval.duration_ms() + 1) as usize;
-            offset_fn = Duration::add_ms;
-        },
+    let duration = match tu {
+        TimeUnit::Nanoseconds => interval.duration_ns(),
+        TimeUnit::Microseconds => interval.duration_us(),
+        TimeUnit::Milliseconds => interval.duration_ms(),
+    };
+    let time_zone_opt_string: Option<String> = match tz {
+        #[cfg(feature = "timezones")]
+        Some(tz) => Some(tz.to_string()),
+        _ => None,
+    };
+    if interval.is_constant_duration(time_zone_opt_string.as_deref()) {
+        // Fast path!
+        return match closed {
+            ClosedWindow::Both => Ok((start..=end)
+                .step_by(duration as usize)
+                .collect::<Vec<i64>>()),
+            ClosedWindow::None => Ok((start..end)
+                .step_by(duration as usize)
+                .skip(1)
```

Comment:
Nit: can we do this skip in the `(start..end)` range directly? That might produce faster code.

Reply:
I couldn't find a way to do this, suggestions I looked for how to exclude the first element resulted in `skip(1)` - do you have a suggestion?

Reply:
The way I did it was `(start+step..end)`.

Reply:
right, indeed 🙏